repair/repair.cc

```diff
@@ -1285,15 +1296,21 @@ future<> repair_service::sync_data_using_repair(
         if (get_repair_module().is_aborted(id.uuid())) {
             throw std::runtime_error("aborted by user request");
         }
+        auto ring_version = erm->get_token_metadata().get_ring_version();
         for (auto shard : boost::irange(unsigned(0), smp::count)) {
-            auto f = container().invoke_on(shard, [keyspace, table_ids, id, ranges, neighbors, reason, ops_info] (repair_service& local_repair) mutable {
+            auto f = container().invoke_on(shard, [keyspace, table_ids, id, ranges, neighbors, reason, ops_info, ring_version] (repair_service& local_repair) mutable {
+                auto erm = local_repair.get_db().local().find_keyspace(keyspace).get_effective_replication_map();
+                auto local_ring_version = erm->get_token_metadata().get_ring_version();
+                if (local_ring_version != ring_version) {
+                    return make_exception_future<>(std::runtime_error("Ring version changed"));
+                }
```

Comment:
Perhaps we'll want to make a sharded<erm>.

Reply:
~= see above.